experiment_prototype/experiment_prototype.py

```diff
@@ -840,24 +967,38 @@ def edit_slice(self, edit_slice_id, **kwargs):
                 errmsg = 'Cannot edit slice: {} not a valid slice parameter'.format(edit_slice_param)
                 raise ExperimentException(errmsg)
 
-        # Move the interface values from old slice to new slice.
-        new_interface_values = {}
-        for ifkey, ifvalue in self._interface.items():
-            if edit_slice_id == ifkey[0]:
-                new_interface_values[ifkey[1]] = ifvalue
-            elif edit_slice_id == ifkey[1]:
-                new_interface_values[ifkey[0]] = ifvalue
+        # Get the interface values of the slice. These are not editable, if
+        # these are wished to be changed add_slice must be used explicitly
+        # to interface a new slice.
+        interface_values = self.get_slice_interfacing(edit_slice_id)
 
-        # This interface value will be removed when the slice is deleted.
-        new_interface_values[edit_slice_id] = None
+        removed_slice = self.del_slice(edit_slice_id)
 
-        # checks are done when slice is added.
-        new_slice_id = self.add_slice(edited_slice, new_interface_values)
+        try:
+            # checks are done on interfacing when slice is added.
+            # interfacing between existing slice_ids cannot be changed after addition.
+            new_slice_id = self.add_slice(edited_slice, interface_values)
+            return new_slice_id
+
+        except ExperimentException:
```

Comment:
yes... there should be some logging done here. It would be nice to move all of the exceptions in add_slice, del_slice, edit_slice to a logging-and-revert function instead of raise in case we are using the experiment update method, to reduce downtimes. I will look into this.